source/modules/MEI.cmn.xml

```diff
@@ -1346,10 +1346,10 @@
     </content>
     <constraintSpec ident="When_not_copyof_beam_content" scheme="isoschematron">
       <constraint>
-        <sch:rule context="mei:beam[not(@copyof)]">
+        <sch:rule context="mei:beam[not(@copyof) and not(@sameas)]">
```

Comment:
```suggestion
        <sch:rule context="mei:beam[not(@copyof or @sameas)]">
```

Reply:
Please be aware that this XPath pattern is setting the context for the following `sch:assert`. This is to say only nodes matching this pattern will be tested against the XPath expression in `sch:assert`, being: 
```xpath 
count(descendant::*[local-name()='note' or local-name()='rest' or local-name()='chord' or local-name()='space']) &gt; 1
```
We only have to test for the presence of at least two child nodes if both attributes (`@copyof` and `@sameas`) are missing.

The following table illustrates the results of the two different XPath patterns in `sch:rule/@ocntext`.

|    | mei:beam[not(@copyof) _and_ not(@sameas)]  | mei:beam[not(@copyof) _or_ not(@sameas)]  |
|:--------|:-------|:-------|
| `<beam/>` |  match   | match   |
| `<beam copyof=""/>`   | no match   | match  |
| `<beam sameas=""/>`   | no match  | match   |
| `<beam copyof="" sameas=""/>`   | no match   |  no match   |

If this table holds `match` in a cell that means, the element will be tested, i.e. child `note`, `rest`, `chord`, and `space` will be counted. An error will be reported if this number is less than `2`.

The big difference in the XPath patterns is the two rows in the middle. While _AND_ would not count their child elements, _OR_ will, because one of the attributes is missing and thus the pattern matches those `beam`s.

Reply:
Sorry. I looked only at the bit that was changed.

Reply:
Same here, didn't realise until now that you changed the parenthesis … Your proposed expression `mei:beam[not(@copyof or @sameas)]`delivers the same results as `mei:beam[not(@copyof) and not(@sameas)]`